### ISSUE TITLE
Method not Implemented error in OpenCV

### ISSUE THREAD

**Issue by AKMANSoft** [bytedeco/javacpp-presets#840]
Im trying to run FastfeatureDetection but its giving me error, that the method is not implemented.  

```
`java.lang.reflect.InvocationTargetException: null
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at image.viewer/com.imviewer.worker.WorkerStub.start(WorkerStub.java:89)
	at image.viewer/com.imviewer.worker.WorkerProcessor.startWorker(WorkerProcessor.java:654)
	at image.viewer/com.imviewer.worker.WorkerProcessor.run(WorkerProcessor.java:645)
Caused by: java.lang.RuntimeException: OpenCV(4.1.2) C:\projects\javacpp-presets\opencv\cppbuild\windows-x86_64\opencv-4.1.2\modules\features2d\src\feature2d.cpp:174: error: (-213:The function/feature is not implemented)  in function 'cv::Feature2D::detectAndCompute

	at org.bytedeco.opencv/org.bytedeco.opencv.opencv_features2d.Feature2D.compute(Native Method)
	at image.viewer/com.imviewer.keypoints.VisualKeypoints.calculateKeyPoints(VisualKeypoints.java:400)
	at image.viewer/com.imviewer.keypoints.VisualKeypoints.<init>(VisualKeypoints.java:101)
	... 7 common frames omitted
````

Thank you

**Comment by saudet** [comment]
Please provide C/C++ code that works!

**Comment by AKMANSoft** [comment]
Im following this example in which is in java scala, but instead of surf im using FastFeatureDetection method

[(https://github.com/bytedeco/javacv-examples/blob/master/OpenCV_Cookbook/src/main/scala/opencv_cookbook/chapter09/Ex7DescribingSURF.scala)]

**Comment by AKMANSoft** [comment]
Are we not supporting OpenCV contributes? like Fast Feature, Surf or Brisk?

**Comment by saudet** [comment]
Let's confirm with a build log such as this one here: https://api.travis-ci.org/v3/job/642185917/log.txt
Yes, it looks like FAST, SURF, and BRISK are getting built. There's most likely just something wrong with your code.

**Comment by AKMANSoft** [comment]
Thank you for confirming, that its build. I have create a test file to replicate the issue, also adding the list of jars I'm adding in pom, in case i miss some thing

```
import static org.bytedeco.opencv.global.opencv_imgcodecs.imread;

import org.bytedeco.opencv.opencv_core.KeyPointVector;
import org.bytedeco.opencv.opencv_core.Mat;
import org.bytedeco.opencv.opencv_features2d.FastFeatureDetector;

public class FastKeyoints {
	
	Mat image;
	Mat descriptor;
	KeyPointVector keypoints;
	
	FastFeatureDetector featureDetector;
	
	public FastKeyoints(String path){
		descriptor = new Mat();
		keypoints = new KeyPointVector();
		featureDetector = FastFeatureDetector.create();
		image = imread(path);
	}
	
	/**
	 * This method extract features from the reference Image
	 */
	public void calculateKeyPoints() {
		featureDetector.detect(image, keypoints);
		featureDetector.compute(image, keypoints, descriptor);
		
		//This also fails and gives same error message. 
		//featureDetector.detectAndCompute(refImage, refImage, refKeypoints, refDescriptor);
	}
	
	public static void main(String[] args) {
		String path = "C:\\Users\\Admin\\Desktop\\Special Tests\\Face Search Demo\\Angelina_9_5.jpg";
		FastKeyoints keypoints = new FastKeyoints(path);
		keypoints.calculateKeyPoints();
		
	}
}
```
. 
```
<javacpp.version>1.5.2</javacpp.version>
                   <dependency>
			<groupId>org.bytedeco</groupId>
			<artifactId>javacv</artifactId>
			<version>${javacpp.version}</version>
		</dependency>

				
		<!-- https://mvnrepository.com/artifact/org.bytedeco/javacpp -->
		<dependency>
			<groupId>org.bytedeco</groupId>
			<artifactId>javacpp</artifactId>
			<version>${javacpp.version}</version>
		</dependency>
		
		<!-- https://mvnrepository.com/artifact/org.bytedeco/opencv -->
		<dependency>
		    <groupId>org.bytedeco</groupId>
		    <artifactId>opencv</artifactId>
		    <version>4.1.2-${javacpp.version}</version>
		    <classifier>windows-x86_64</classifier>
		</dependency>
		
		<!-- https://mvnrepository.com/artifact/org.bytedeco/opencv-platform -->
		<dependency>
		    <groupId>org.bytedeco</groupId>
		    <artifactId>opencv-platform</artifactId>
		    <version>4.1.2-${javacpp.version}</version>
		</dependency>
	
		<!-- https://mvnrepository.com/artifact/org.bytedeco/openblas -->
		<dependency>
		    <groupId>org.bytedeco</groupId>
		    <artifactId>openblas</artifactId>
		    <version>0.3.7-${javacpp.version}</version>
		    <classifier>windows-x86_64</classifier>
		</dependency>
```

**Comment by AKMANSoft** [comment]
Update i have test the above code with Brish and Surf, they are working, but Fast is not working only .

**Comment by saudet** [comment]
Could you try again with OpenCV 4.2.0 to make sure the issue is still present there?

**Comment by AKMANSoft** [comment]
Yes I have tested it now with OpenCV 4.2.0 using Snapshot version. Error still remaining's

```
Exception in thread "main" java.lang.RuntimeException: OpenCV(4.2.0) C:\projects\javacpp-presets\opencv\cppbuild\windows-x86_64\opencv-4.2.0\modules\features2d\src\feature2d.cpp:174: error: (-213:The function/feature is not implemented) in function 'cv::Feature2D::detectAndCompute'

at org.bytedeco.opencv.opencv_features2d.Feature2D.compute(Native Method)
at com.imviewer.panes.FastKeyoints.calculateKeyPoints(FastKeyoints.java:41)
at com.imviewer.panes.FastKeyoints.main(FastKeyoints.java:53)
```

**Comment by Karl515** [comment]
I'm also getting the error just by running the sample [](https://github.com/bytedeco/javacv)
```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:464)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:363)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:1051)
Caused by: java.lang.RuntimeException: OpenCV(4.1.2) C:\projects\javacpp-presets\opencv\cppbuild\windows-x86_64\opencv-4.1.2\modules\core\src\persistence.cpp:719: error: (-49:Unknown error code -49) Input file is invalid in function 'cv::FileStorage::Impl::open'

	at org.bytedeco.opencv.opencv_objdetect.CascadeClassifier.allocate(Native Method)
	at org.bytedeco.opencv.opencv_objdetect.CascadeClassifier.<init>(CascadeClassifier.java:56)
	at main.Main.main(Main.java:66)
	... 11 more
Exception running application main.Main
```

**Comment by saudet** [comment]
Here's the answer:
https://stackoverflow.com/questions/35256174/opencv-feature-fast-not-implemented-in-source-code
**tl;dr**: We can't call `compute()` or `detectAndCompute()` with that algorithm since it's not implemented.
